ratelimit.go

```diff
@@ -0,0 +1,66 @@
+package strava
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var RateLimitLast RateLimit
+
+type RateLimit struct {
+	LastRequestTime time.Time
+	LimitShort      int
+	LimitLong       int
+	UsageShort      int
+	UsageLong       int
+}
+
+// returns true if rate limit was reached during last X seconds
+func RateLimitReachedDuringLast(seconds int64) bool {
```

Comment:
How about renaming `RateLimitLast` to `RateLimiting` then define this method on the rate limit struct and call it `Exceeded`.  Then you could do something like `if !strava.RateLimiting.Exceeded()`. 

I think this method should use the information in the RateLimit struct along with the current time to figure out if the rate is exceeded, something like:

```
now := time.Now().Unix()
req := LastRequestTime.Unix()

// same 15 minutes block
if now / 900 == req /900 && UsageShort > LimitShort {
  return false
} 

// same day
if now / 86400 == req / 86400 && UsageLong > LimitLong {
  return false
}

return true